app/models/student.rb

```diff
@@ -23,6 +23,8 @@ class Student < ApplicationRecord
 
   validates :uniqname, uniqueness: { scope: :program, message: "is already in the program list" }
 
+  scope :eligible_drivers, -> { where.not(class_training_date: nil) }
```

Comment:
Wait until you use a call to this before adding it the codebase

Reply:
I did call it in the reservation_controller in the add_drivers. You can choose drivers only from eligible_drivers. 

```
def add_drivers
    @drivers = @reservation.program.students.eligible_drivers
    unless is_admin?(current_user)
      driver = Student.find_by(program_id: @reservation.program_id, uniqname: current_user.uniqname)
      @reservation.update(driver_id: driver.id)
    end
end
```
I created a ticket to write the code for that scope. I did exactly what you suggested before - create placeholders.

Reply:
The :eligible_drivers scope is defined in LRA-595